lib/azure/armrest/configuration.rb

```diff
@@ -193,7 +189,10 @@ def validate_token_time(time)
       # a non-preview version that is not set in the future. Otherwise, just
       # just the most recent one.
       #
-      def set_provider_info(providers)
+      def set_provider_api_versions
+        # A lookup table for getting api-version strings per provider and service.
+        @provider_api_versions = Hash.new { |hash, key| hash[key] = {} }
```

Comment:
No need. `set_provider_info` is called by the constructor anyway. Technically if this method is called separately (which does not happen), `set_provider_info` needs to be reset here.